### PR TITLE
Add parameter to pass role ARN to `GlueJobOperator `

### LINKED ISSUE #33248: GlueOperator: iam_role_arn as a parameter
### Description
Hi,
There is mandatory parameter iam_role_name parameter for GlueJobOperator/GlueJobHook.
It adds additional step of translating it to the arn, which needs  connectivity to the global iam AWS endpoint (no privatelink availabale).
For private setups it needs opening connectivity + proxy configuration to make it working.  
It would be great to have also possibility to just pass directly iam_role_arn and avoid this additional step.
### Use case/motivation

Role assignation does not need external connectivity, possibility of adding arn instead of the name.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by erdos2n**
GlueOperator currently requires `iam_role_name` and `endpoint_url` to fetch the ARN from AWS. This ARN eventually gets passed into the config for Glue Job. 

Resolves #33248

### CODE REVIEW

**Comment by erdos2n** [comment]
@eladkal , I added test cases for the `role_arn`, can you let me know if the tests were added correctly? 
Thank you

File: airflow/providers/amazon/aws/hooks/glue.py [hunk 1]
```diff
@@ -44,7 +44,8 @@ class GlueJobHook(AwsBaseHook):
     :param retry_limit: Maximum number of times to retry this job if it fails
     :param num_of_dpus: Number of AWS Glue DPUs to allocate to this Job
     :param region_name: aws region name (example: us-east-1)
-    :param iam_role_name: AWS IAM Role for Glue Job Execution
+    :param iam_role_name: AWS IAM Role for Glue Job Execution. If set iam_role_arn must equal None.
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/hooks/glue.py]:
```suggestion
    :param iam_role_name: AWS IAM Role for Glue Job Execution. If set `iam_role_arn`, must equal None.
```

File: airflow/providers/amazon/aws/hooks/glue.py [hunk 2]
```diff
@@ -44,7 +44,8 @@ class GlueJobHook(AwsBaseHook):
...
+    :param iam_role_arn: AWS IAM Role ARN for Glue Job Execution, If set iam_role_name must equal None.
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/hooks/glue.py]:
```suggestion
    :param iam_role_arn: AWS IAM Role ARN for Glue Job Execution, If set `iam_role_name`, must equal None.
```

File: airflow/providers/amazon/aws/hooks/glue.py [hunk 3]
```diff
@@ -114,12 +119,14 @@ def create_glue_job_config(self) -> dict:
             "ScriptLocation": self.script_location,
         }
         command = self.create_job_kwargs.pop("Command", default_command)
-        execution_role = self.get_iam_execution_role()
+        if not self.role_arn:
+            execution_role = self.get_iam_execution_role()
+            self.role_arn = execution_role["Role"]["Arn"]
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/hooks/glue.py]: nit. There is no need to save it as instance variable, local variable is enough
```suggestion
            role_arn = execution_role["Role"]["Arn"]
```

File: airflow/providers/amazon/aws/operators/glue.py [hunk 4]
```diff
@@ -53,7 +53,8 @@ class GlueJobOperator(BaseOperator):
     :param num_of_dpus: Number of AWS Glue DPUs to allocate to this Job.
     :param region_name: aws region name (example: us-east-1)
     :param s3_bucket: S3 bucket where logs and local etl script will be uploaded
-    :param iam_role_name: AWS IAM Role for Glue Job Execution
+    :param iam_role_name: AWS IAM Role for Glue Job Execution. If set iam_role_arn must equal None.
+    :param iam_role_arn: AWS IAM ARN for Glue Job Execution. If set iam_role_name must equal None.
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/operators/glue.py]:
```suggestion
    :param iam_role_name: AWS IAM Role for Glue Job Execution. If set `iam_role_arn`, must equal None.
    :param iam_role_arn: AWS IAM ARN for Glue Job Execution. If set `iam_role_name`, must equal None.
```

File: tests/providers/amazon/aws/hooks/test_glue.py [hunk 5]
```diff
@@ -90,6 +90,56 @@ class JobNotFoundException(Exception):
         assert result is False
         mock_conn.get_job.assert_called_once_with(JobName=job_name)
 
+    @mock.patch.object(GlueJobHook, "get_iam_execution_role")
+    @mock.patch.object(AwsBaseHook, "conn")
+    def test_role_arn_has_job_exists(self, mock_conn, mock_get_iam_execution_role):
```
Comment (vincbeck) [inline tests/providers/amazon/aws/hooks/test_glue.py:111]: Maybe add a test to test that if both parameters are provided, you get a `ValueError`?

File: airflow/providers/amazon/aws/hooks/glue.py (hunk 1 above)
Comment (erdos2n) [reply airflow/providers/amazon/aws/hooks/glue.py]: @vincbeck updated

File: airflow/providers/amazon/aws/hooks/glue.py (hunk 2 above)
Comment (erdos2n) [reply airflow/providers/amazon/aws/hooks/glue.py]: @vincbeck updated

File: airflow/providers/amazon/aws/hooks/glue.py (hunk 3 above)
Comment (erdos2n) [reply airflow/providers/amazon/aws/hooks/glue.py]: @vincbeck updated. assigned the role for both cases to a local variable `role_arn`. Let me know if this works. I did not want to put the if/else logic on line #131.

File: airflow/providers/amazon/aws/operators/glue.py (hunk 4 above)
Comment (erdos2n) [reply airflow/providers/amazon/aws/operators/glue.py]: @vincbeck updated

File: tests/providers/amazon/aws/hooks/test_glue.py (hunk 5 above)
Comment (erdos2n) [reply tests/providers/amazon/aws/hooks/test_glue.py:111]: @vincbeck I've added a test case, let me know if any changes are required.

File: tests/providers/amazon/aws/hooks/test_glue.py [hunk 6]
```diff
@@ -66,6 +66,22 @@ def test_get_iam_execution_role(self, role_path):
         assert "Arn" in iam_role["Role"]
         assert iam_role["Role"]["Arn"] == f"arn:aws:iam::123456789012:role{role_path}{expected_role}"
 
+    @mock.patch.object(GlueJobHook, "get_iam_execution_role")
+    @mock.patch.object(GlueJobHook, "conn")
+    def test_init_iam_role_value_error(self, mock_conn, mock_get_iam_execution_role):
```
Comment (erdos2n) [inline tests/providers/amazon/aws/hooks/test_glue.py:71]: @vincbeck here is the new test case for the `ValueError`

**Review by vincbeck — APPROVED** [review]
LGTM

**Comment by vincbeck** [comment]
You have some static checks failure, please [run them](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst) to fix these failures.

**Pull request merged**